src/k8s/api/v1/types.go

```diff
@@ -0,0 +1,100 @@
+package v1
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+type BootstrapConfig struct {
+	// Components are the components that should be enabled on bootstrap.
+	Components []string
+	// ClusterCIDR is the CIDR of the cluster.
+	ClusterCIDR string
+}
+
+// SetDefaults sets the fields to default values.
+func (b *BootstrapConfig) SetDefaults() {
+	b.Components = []string{"dns", "network"}
+	b.ClusterCIDR = "10.1.0.0/16"
+}
+
+// ToMap converts the BootstrapConfig to a map of strings.
+// TODO: should we move those helpers to utils?
+func (b *BootstrapConfig) ToMap() map[string]string {
+	return map[string]string{
+		"components":   strings.Join(b.Components, ","),
+		"cluster-cidr": b.ClusterCIDR,
+	}
```

Comment:
I would recommend to have this in a single `"bootstrapConfig"` entry, then format as JSON or YAML. This way we can perform better validations on each key, or easily refactor the config format in the future.